### ISSUE TITLE
Not converting to typst math when equation label is included

### ISSUE THREAD

**Issue by jzelner** [quarto-dev/quarto-cli#7744]
### Bug description

When I run this equation from qmd -> typst -> pdf, I get the expected behavior when omitting an equation label:

Here is the original TeX math in Quarto:

<img width="571" alt="image" src="https://github.com/quarto-dev/quarto-cli/assets/136323/2771361a-ad8a-4f05-9933-d9aad39d7c0c">

And this is the correctly generated typst math:

<img width="556" alt="image" src="https://github.com/quarto-dev/quarto-cli/assets/136323/5d6e574b-9740-4271-9379-e259a11b4e5d">

Now, re-adding the equation label (this compiles properly to PDF using xelatex and to HTML):

<img width="554" alt="image" src="https://github.com/quarto-dev/quarto-cli/assets/136323/d4970dbb-8600-42f7-95dc-8e76d6f17a73">

I get the raw TeX math in the typst intermediate document and it will not compile:

<img width="561" alt="image" src="https://github.com/quarto-dev/quarto-cli/assets/136323/042f387a-8980-4641-a85a-4032c4a96f07">

### Steps to reproduce

$$
\begin{aligned}
    \frac{dS_1}{dt} &= -\frac{\rho \beta S_1(c_{11} I_1 + c_{12} I_2)}{N} \\
    \frac{dS_2}{dt} &= -\frac{\beta S_2(c_{12} I_1 + c_{22} I_2)}{N} \\
    \frac{dI_1}{dt} &= \frac{ \rho \beta S_1(c_{11} I_1 + c_{12} I_2)}{N} - \gamma I_1 \\
    \frac{dI_2}{dt} &= \frac{\beta S_2(c_{12} I_1 + c_{22} I_2)}{N} - \gamma I_2 \\
    \frac{dR_1}{dt} &= \gamma I_1 \\
    \frac{dR_2}{dt} &= \gamma I_2 
 \end{aligned} 
$$ {#eq-model}

**Comment by cscheid** [comment]
I can repro this locally, FWIW. It's a bug in how we're annotating equations. The code we have only works for inline equations, and not block equations.

**Comment by jzelner** [comment]
This one is almost working, but in a document with multiple numbered equations, quarto is emitting the numbered reference rather than the reference style as required by typst. For example, here is an equation from my document which was assigned the number 5:

<img width="650" alt="image" src="https://github.com/quarto-dev/quarto-cli/assets/136323/058b04e5-e61c-4035-b66e-1afcf94826ba">

This won't compile using `typst compile`, but if I find and replace all instances a numbered reference > 1 using the regex `\(\d\)`with `(1)` it compiles and gives the appropriate output, correctly numbered, below: 

<img width="579" alt="image" src="https://github.com/quarto-dev/quarto-cli/assets/136323/b9924a88-76d4-41d8-93b6-c7b51a26f9d8">

So, basically it looks like quarto is doing a bit too much ehre and just needs to supply the numbering *style* rather than the number itself to typst. Thanks!

**Comment by cscheid** [comment]
I misinterpreted the behavior of the parameter 🤦. This is fixed on `main` now. Thanks for following up!
